mysql-test/t/acl_mdl_scalability.test

```diff
@@ -0,0 +1,255 @@
+      printf("In main: creating thread %ld\n", t);
+      rc = pthread_create(&threads[t], NULL, ThreadTask, (void *)t);
+      if (rc){
+         printf("ERROR; return code from pthread_create() is %s\n", strerror(rc));
+         exit(-1);
+      }
+   }
+
+   for(long t=0; t<NUM_THREADS; t++){
+	pthread_join(threads[t], NULL);
+   }
+
+   printf("\n\nfinish executing main\n\n");
+   return(0);
+}
+EOF
+--echo # Require g++ to compile this program. If the exec fails "sh: 1: g++: not found". Please install g++ and retry
+--exec g++ $MYSQL_TMP_DIR/thread_limit_test.cc -lpthread -o $MYSQL_TMP_DIR/thread_limit 2>&1 > $MYSQL_TMP_DIR/thread_limit_gcc_error.txt
+
+--echo # if the below exec fails, it means you the operating system has limit of threads less than 12K
+--echo # This testcase needs OS that can create 12k threads.
+--echo # Try these settings (only if they are less 12,000)
+--echo # ulimit -i  12000
+--echo # echo 12000 > /proc/sys/kernel/threads-max
+--echo # echo 60000 > /proc/sys/vm/max_map_count
+--echo # echo 12000 > /proc/sys/kernel/pid_max
+--echo # echo 12000 > /sys/fs/cgroup/pids/user.slice/user-$UID.slice/pids.max
+--exec $MYSQL_TMP_DIR/thread_limit 2>&1 > $MYSQL_TMP_DIR/thread_limit_gcc_error.txt
+--remove_file $MYSQL_TMP_DIR/thread_limit_test.cc
+--remove_file $MYSQL_TMP_DIR/thread_limit_gcc_error.txt
+--remove_file $MYSQL_TMP_DIR/thread_limit
+
+--echo # Precondition check of system over. System can handle 12K concurrent threads
+
+create user satya@'%' identified by 'hello';
+grant all  on  test.* to satya@'%';
+grant all  on  mysql.* to satya@'%';
+
+select @@global.max_connections;
+
+--connect (con_flush,localhost,root,,)
+--connect (con0,localhost,root,,)
+
+--let $i = 1
+while ($i <= 200) {
+ --connect (con$i, localhost, root)
+ --inc $i
+}
+
+--let $i = 201
+while ($i <= 700) {
+ --connect (con$i, 127.0.0.1, satya,hello,test)
+ --inc $i
+}
+
+--connection default
+--let $initial_con_count=`SELECT COUNT(*) FROM information_schema.processlist`
+
+--echo # Sleeping 15 seconds for allowing user to connect client for observability
+--echo # we cannot establish connection later as database freezes. Do now if you wish
+--echo # be in mysql-source/bld/mysql-test directory and execute the below command
+--echo # ../runtime_output_directory/mysql -uroot -S ./var/tmp/mysqld.1.sock
+#--sleep 15
+#--echo # sleep over
+
+--echo # connection con0: Hold ACL x lock and let readers wait
+--connection con0
+SET DEBUG_SYNC= 'acl_x_lock SIGNAL x_locked WAIT_FOR all_wait_s_locks_done';
+--send FLUSH PRIVILEGES
+
+--echo # 200 connections try for SHARED ACL lock and go into waiting list
+--echo # because FLUSH PRIVILEGS hold EXCLUSIVE lock
+--let $i = 1
+while ($i <= 200) {
+ --echo # Connection con$i
+ --connection con$i
+ SET DEBUG_SYNC = 'now WAIT_FOR x_locked NO_CLEAR_EVENT';
+ SET DEBUG_SYNC = 'acl_s_lock SIGNAL s_locked WAIT_FOR new_xlock NO_CLEAR_EVENT';
+ --send SHOW CREATE USER root@localhost
+ --inc $i
+}
```

Comment:
Not a strong opinion though! These statements seem to flood the result.file. If it makes sense, can we place this loop between the --disable_query_log and --enable_query_log?